docs/ethsimulatev1-notes.md

```diff
@@ -0,0 +1,202 @@
+# eth_simulate
+This document contains some extra information that couldn't be fit to the specification document directly.
+
+## Default block values
+Unlike `eth_call`, `eth_simulateV1`'s calls are conducted inside blocks. We don't require user to define all the fields of the blocks so here are the defaults that are assumed for blocks parameters:
+
+| parameter name | default value |
+-----------------|-----------------------
+| prevRandao | `0x0000000000000000000000000000000000000000000000000000000000000000` |
+| feeRecipient | `0x0000000000000000000000000000000000000000` |
+| mixHash | `0x0000000000000000000000000000000000000000000000000000000000000000` |
+| nonce | `0x0` |
+| extraData | `0x0000000000000000000000000000000000000000000000000000000000000000` |
+| difficulty | The same as the base block defined as the second parameter in the call |
+| gasLimit | The same as the base block defined as the second parameter in the call |
+| hash | Calculated normally |
+| parentHash | Previous blocks hash |
+| timestamp | The timestamp of previous block + `network block time` (12s on Ethereum Mainnet) |
+| baseFeePerGas | When validation mode is true, baseFeePerGas is calculated on what it should be according to Ethereum's spec. When validation mode is false, the baseFeePerGas is set to zero |
+| sha3Uncles | Empty trie root |
+| withdrawals | Empty array |
+| uncles | Empty array |
+| blobBaseFee | When validation mode is true, blobBaseFee is calculated on what it should be according to EIP-4844 spec. When validation mode is false, the blobBaseFee is set to zero |
+| number | Previous block number + 1 |
+| logsBloom | Calculated normally. ETH logs are not part of the calculation |
+| receiptsRoot | Calculated normally |
+| transactionsRoot | Calculated normally |
+| size | Calculated normally |
+| withdrawalsRoot | Calculated normally |
+| gasUsed | Calculated normally |
+| stateRoot | Calculated normally |
+
+## Default values for transactions
+As eth_simulate is an extension to `eth_call` we want to enable the nice user experience that the user does not need to provide all required values for a transaction. We are assuming following defaults if the variable is not provided by the user:
+| parameter name | description |
+-----------------|-----------------------
+| type | `0x2` |
+| nonce | Take the correct nonce for the account prior eth_simulate and increment by one for each transaction by the account |
+| to | `null` |
+| from | `0x0000000000000000000000000000000000000000` |
+| gasLimit | (blockGasLimit - SumOfGasLimitOfTransactionsWithDefinedGasLimit) / NumberOfTransactionsWithoutKnownGasLimit |
```

Comment:
NMC, reth and geth _call_ this field `gas` not `gasLimit`.
Thus, I  think it makes more sense to rename it to `gas` here :)

Reply:
I like `gasLimit` because it makes it more clear for novice developers.  For most new developers, when they start working on Ethreum `gas` is a confusing concept.  Calling the variables `gasLimit` and `gasPrice` helps provide clarity where `gas` and `gasPrice` is a bit more confusing.

Reply:
Hey, thanks for your thoughts on this. Initially I agreed 100%. 

My 2c:
I have had some spare time so I decided to open PRs to NMC/geth/reth to fix this,  only to release that all 3 clients have their own version of _TransactionRPC_ - an object to handle ser/de of transactions from/to all RPC calls.

This means that if in `eth_Simulate` name `gasLimit` is used instead of `gas` it will be inconsistent with all other RPC calls (e.g. `get_Transaction/send_Transaction`).
IMO this not good DX/user friendly

Reply:
That is a good argument, and puts us in a difficult position because on the one hand we should want to care about new dev experience for people integrating with Ethereum, and the expectation is that `eth_simulate` will effectively supercede `eth_call` over time, so it is a good opportunity to make the experience better.  On the other hand, we don't necessarily want to make building/maintaining a client harder.